File: scripts/fix-crlf.py
#!/usr/bin/env python3
from pathlib import Path

root = Path(__file__).resolve().parent.parent
for pattern in ("scripts/*.sh", ".env", ".env.example"):
    for path in root.glob(pattern):
        text = path.read_text(encoding="utf-8")
        fixed = text.replace("\r\n", "\n").replace("\r", "\n")
        if fixed != text:
            path.write_text(fixed, encoding="utf-8", newline="\n")
            print(f"fixed: {path.relative_to(root)}")

File: scripts/fix-crlf.pl
#!/usr/bin/env perl
use strict;
use warnings;
for my $path (glob("scripts/*.sh")) {
  open my $fh, "<:raw", $path or die $path;
  my $data = do { local $/; <$fh> };
  close $fh;
  my $fixed = $data;
  $fixed =~ s/\r\n/\n/g;
  $fixed =~ s/\r/\n/g;
  next if $fixed eq $data;
  open my $out, ">:raw", $path or die $path;
  print {$out} $fixed;
  close $out;
  print "fixed $path\n";
}
